sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -76,14 +76,15 @@ case class CreateFunctionCommand(
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
     val func = CatalogFunction(FunctionIdentifier(functionName, databaseName), className, resources)
+    catalog.loadFunctionResources(resources)
+    // We fail fast if function class is not exists.
+    catalog.requireFunctionClassExists(func)
```

Comment:
is this needed? shouldn't `loadFunctionResources` already does the work?

Reply:
`loadFunctionResources` only check the resource is exists, e.g. jar / file. Not include function class.

Reply:
can we register the function when creating it? That should load the class.

Reply:
For a permanent function, does it better cache when query on it? Seems the most use case is as a Hive client.